handler/src/main/java/io/netty/handler/pcap/PCapWriter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.pcap;
+
+import io.netty.buffer.ByteBuf;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+final class PCapWriter implements Closeable {
+    private final SimpleDateFormat sdf = new SimpleDateFormat("SSSSSSSSS");
+    private final OutputStream outputStream;
+
+    /**
+     * This uses {@link OutputStream} for writing Pcap.
+     * Pcap Global Header is also written on construction.
+     *
+     * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
+     */
+    PCapWriter(OutputStream outputStream, ByteBuf byteBuf) throws IOException {
+        this.outputStream = outputStream;
+
+        PcapHeaders.writeGlobalHeader(byteBuf);
+        byteBuf.readBytes(outputStream, byteBuf.readableBytes());
+        byteBuf.release();
+    }
+
+    /**
+     * Write Packet in Pcap OutputStream.
+     *
+     * @param packetHeaderBuf Packer Header {@link ByteBuf}
+     * @param packet          Packet
+     * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
+     */
+    void writePacket(ByteBuf packetHeaderBuf, ByteBuf packet) throws IOException {
+        long currentTime = System.currentTimeMillis();
+
+        String microsecond = sdf.format(new Date(currentTime));
+        microsecond = microsecond.substring(microsecond.indexOf(".") + 1);
+
+        PcapHeaders.writePacketHeader(
+                packetHeaderBuf,
+                currentTime,
+                Integer.parseInt(microsecond),
+                packet.readableBytes(),
+                packet.readableBytes()
+        );
+
+        packetHeaderBuf.readBytes(outputStream, packetHeaderBuf.readableBytes());
+        packet.readBytes(outputStream, packet.readableBytes());
+        packetHeaderBuf.release();
```

Comment:
I would remove this and let the caller handle this. This is also more consistent as you also don't release the `packet` here